src/app/games/diaballik/DiaballikState.ts

```diff
@@ -0,0 +1,36 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { GameStateWithTable } from 'src/app/jscaip/GameStateWithTable';
+import { Player, PlayerOrNone } from 'src/app/jscaip/Player';
+import { ComparableObject } from 'src/app/utils/Comparable';
+import { TableUtils } from '../../utils/ArrayUtils';
+
+export class DiaballikPiece implements ComparableObject {
+
+    public static readonly NONE: DiaballikPiece = new DiaballikPiece(PlayerOrNone.NONE, false);
+    public static readonly ZERO: DiaballikPiece = new DiaballikPiece(Player.ZERO, false);
+    public static readonly ZERO_WITH_BALL: DiaballikPiece = new DiaballikPiece(Player.ZERO, true);
+    public static readonly ONE: DiaballikPiece = new DiaballikPiece(Player.ONE, false);
+    public static readonly ONE_WITH_BALL: DiaballikPiece = new DiaballikPiece(Player.ONE, true);
+
+    private constructor(public readonly owner: PlayerOrNone,
+                        public readonly holdsBall: boolean)
+    {
+    }
+
+    public equals(other: this): boolean {
+        return this === other;
+    }
+}
+
+export class DiaballikState extends GameStateWithTable<DiaballikPiece> {
+
+    private static readonly SIZE: number = 7;
+
+    public static isOnBoard(coord: Coord): boolean {
+        return coord.isInRange(DiaballikState.SIZE, DiaballikState.SIZE);
+    }
```

Comment:
KILL THE POOP OUT OF THOSE !
Pratique dépréciée !

Reply:
quoi ?  `public static isOnBoard` apparait dans 19 `BliBliState.ts`, c'pas déprécié du tout lo !

Reply:
c'est intégralement du pré-configurate, Diaballik n'est pas du tout pré-configurate vu que contemporain-à-configurate, donc c'est non non non !